sktime/split/expandinggreedy.py

```diff
@@ -16,21 +17,25 @@
 class ExpandingGreedySplitter(BaseSplitter):
     """Splitter that uses all available data.
 
-    Takes an integer `test_size` that defines the number of steps included in the
+    Takes an integer ``test_size`` that defines the number of steps included in the
     test set of each fold. The train set of each fold will contain all data before
-    the test set. If the data contains multiple instances, `test_size` is
+    the test set. If the data contains multiple instances, ``test_size`` is
     _per instance_.
 
-    If no `step_length` is defined, the test sets (one for each fold) will be
-    adjacent, taken from the end of the dataset.
+    If no ``step_length`` is defined, the test sets (one for each fold) will be
+    adjacent and disjoint, taken from the end of the dataset.
 
-    For example, with `test_size=7` and `folds=5`, the test sets in total will cover
+    For example, with ``test_size=7`` and ``folds=5``, the test sets in total will cover
     the last 35 steps of the data with no overlap.
 
     Parameters
     ----------
-    test_size : int
-        The number of steps included in the test set of each fold.
+    test_size : int or float
+        If int: the number of steps included in the test set of each fold.
+            Formally, steps are consecutive ``iloc`` indices.
+        If float: the proportion of steps included in the test set of each fold,
+            as a proportion of the total number of index values.
+            Cave: not the ``loc`` proportion between start and end.
```

Comment:
Sorry, I don't follow this caveat. Can you please explain?

Reply:
If the indices are, say, `[1, 2, 3, 4, 100]`, then the last 40% by `loc` would be only `100` (everything at or above 60), but by `iloc` it would be `[4, 100]` (everything at or above the 4-th index)